src/libraries/System.Runtime.Caching/src/System/Runtime/Caching/MemoryCacheEntry.cs

```diff
@@ -144,7 +144,10 @@ internal DateTime UtcLastUpdateUsage
 
             _callback = removedCallback;
 
-            if (dependencies != null)
+            // CacheItemPolicy.ChangeMonitors is frequntly the source for 'dependencies', and that property
+            // is never null. So check that the collection of dependencies is not empty before allocating
+            // the 'seldom' used fields.
+            if (dependencies != null && dependencies.Count > 0)
             {
```

Comment:
I think there's a bug here. The dependencies param passed originally comes from CacheItemPolicy.ChangeMonitors. The [docs](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.caching.cacheitempolicy.changemonitors?view=dotnet-plat-ext-5.0#remarks) for that property says:
> If you change the set of change monitors on a CacheItemPolicy object after the CacheItemPolicy object has been passed to an ObjectCache implementation, the changes have no effect.

This implies that the dependencies are expected to be copied when passed in. If a caller passes a CacheItemPolicy to MemoryCache.Set, and then modifies the CacheItemPolicy.ChangeMonitors collection afterwards, that change will be reflected in the collection stored here. It's also not thread safe. For example, CallNotifyOnChanged gets an iterator which will throw if the collection is modified while iterating.

Reply:
Good catch @mconnew. That issue has been present since NetFx days and quite frankly won't change in NetFx. I've opened a new issue for it here. #52033 

As far as this SeldomUsedFields fix goes... it looks fine?

Reply:
Yes, for the scope of this fix, it looks good.